### PR TITLE
Optimize `inbounds` functions

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

```SUMMARY: None```

#### Purpose of change:

First step to optimize `inbounds` functions and move them into one place instead of scattered functions and macros.

#### Additional context:

See commit comments.

### CODE REVIEW

File: src/enums.h [hunk 1]
```diff
@@ -314,9 +314,37 @@ inline bool operator<( const tripoint &a, const tripoint &b )
...
 static constexpr tripoint tripoint_zero { 0, 0, 0 };
+static constexpr tripoint tripoint_max{ INT_MAX, INT_MAX, INT_MAX };
 
 static constexpr point point_min{ tripoint_min.x, tripoint_min.y };
 static constexpr point point_zero{ tripoint_zero.x, tripoint_zero.y };
+static constexpr point point_max{ tripoint_max.x, tripoint_max.y };
+
+/** Checks if given tripoint is inbounds of given min and max tripoints using given clearence **/
+inline bool generic_inbounds( const tripoint &p,
+                              const tripoint &p_min = tripoint_min,
```
Comment (BevapDin) [inline src/enums.h]: A default for min and max seems pointless. What would be the meaning of `generic_inbounds( p )`? What range does that check?

File: src/editmap.cpp [hunk 2]
```diff
@@ -41,20 +41,16 @@
 #include <vector>
 
 #define dbg(x) DebugLog((DebugLevel)(x),D_GAME) << __FILE__ << ":" << __LINE__ << ": "
-#define maplim 132
-#define pinbounds(p) ( p.x >= 0 && p.x < maplim && p.y >= 0 && p.y < maplim)
+
+static constexpr tripoint editmap_limit_min( 0, 0, -OVERMAP_DEPTH );
```
Comment (BevapDin) [inline src/editmap.cpp]: Editmap limits are the same as for the main `map` class because it uses such a map instance.

File: src/enums.h (hunk 1 above)
Comment (ZhilkinSerg) [reply src/enums.h]: You are correct. It is still WIP, but I forgot to add tag (fixed it now).

File: src/editmap.cpp (hunk 2 above)
Comment (ZhilkinSerg) [reply src/editmap.cpp]: That is kind of temporary transitional variables while I changing the code to use new functions. There are would be have several types of similar limits and I plan to put all of them to one place eventually.

File: src/editmap.cpp [hunk 3]
```diff
@@ -41,20 +41,16 @@
...
+static constexpr tripoint editmap_limit_max( EDIT_MAP_LIMIT, EDIT_MAP_LIMIT, OVERMAP_HEIGHT );
+static constexpr tripoint editmap_clearence_min( tripoint_zero );
```
Comment (jbytheway) [inline src/editmap.cpp]: s/clearence/clearance/

File: src/enums.h [hunk 4]
```diff
@@ -314,9 +314,37 @@ inline bool operator<( const tripoint &a, const tripoint &b )
...
+                              const tripoint &p_max = tripoint_max,
+                              const tripoint &clearence_min = tripoint_zero,
+                              const tripoint &clearence_max = tripoint_zero )
```
Comment (jbytheway) [inline src/enums.h]: To reduce the number of arguments we could have a type for a box (rectangle) which is stored as two tripoints (points) for the min and max corners of the box.

Not sure whether this single use warrants that, but wanted to raise the possibility.

Comment (ZhilkinSerg) [reply src/enums.h]: Good point.

**Pull request merged**